code/modules/power/reactor/reactor.dm

```diff
@@ -0,0 +1,1165 @@
+			if(R.z == usr.z)
+				reactor = R
+				break
+	active = TRUE
+
+/datum/computer_file/program/nuclear_monitor/kill_program(forced = FALSE)
+	active = FALSE
+	..()
+
+/datum/computer_file/program/nuclear_monitor/ui_data()
+	var/list/data = get_header_data()
+	data["powerData"] = powerData
+	data["kpaData"] = kpaData
+	data["tempCoreData"] = tempCoreData
+	data["tempInputData"] = tempInputData
+	data["tempOutputData"] = tempOutputData
+	data["coreTemp"] = reactor ? reactor.temperature : 0
+	data["coolantInput"] = reactor ? reactor.last_coolant_temperature : 0
+	data["coolantOutput"] = reactor ? reactor.last_output_temperature : 0
+	data["power"] = reactor ? reactor.last_power_produced : 0
+	data["kpa"] = reactor ? reactor.pressure : 0
+	data["k"] = reactor ? reactor.K : 0
+	return data
+
+/datum/computer_file/program/nuclear_monitor/ui_act(action, params)
+	if(..())
+		return TRUE
+
+	switch(action)
+		if("swap_reactor")
+			var/list/choices = list()
+			for(var/obj/machinery/atmospherics/components/trinary/nuclear_reactor/R in GLOB.machines)
+				if(R.z != usr.z)
+					continue
+				choices += R
+			reactor = input(usr, "What reactor do you wish to monitor?", "[src]", null) as null|anything in choices
+			powerData = list()
+			kpaData = list()
+			tempCoreData = list()
+			tempInputData = list()
+			tempOutputData = list()
+			return TRUE
+
+/obj/effect/decal/nuclear_waste
+	name = "plutonium sludge"
+	desc = "A writhing pool of heavily irradiated, spent reactor fuel. You probably shouldn't step through this..."
+	icon = 'icons/obj/machines/reactor_parts.dmi'
+	icon_state = "nuclearwaste"
+	alpha = 150
+	light_color = LIGHT_COLOR_GREEN
+	color = "#ff9eff"
+
+/obj/effect/decal/nuclear_waste/Initialize(mapload)
+	. = ..()
+	for(var/obj/A in get_turf(src))
+		if(istype(A, /obj/structure))
+			qdel(src) //It is more processing efficient to do this here rather than when searching for available turfs.
+	set_light(1)
+	AddComponent(/datum/component/radioactive, 1000, src, 0)
+
+/obj/effect/landmark/nuclear_waste_spawner //Clean way of spawning nuclear gunk after a reactor core meltdown.
+	name = "Nuclear waste spawner"
+	var/range = 15 //15 tile radius to spawn goop
+
+/obj/effect/landmark/nuclear_waste_spawner/strong
+	range = 30
+
+/obj/effect/landmark/nuclear_waste_spawner/proc/fire()
+	playsound(loc, 'sound/effects/gib_step.ogg', 100)
+	for(var/turf/open/floor in orange(range, get_turf(src)))
+		if(prob(35)) //Scatter the sludge, don't smear it everywhere
+			new /obj/effect/decal/nuclear_waste (floor)
+	qdel(src)
+
+/obj/effect/decal/nuclear_waste/Crossed(atom/movable/AM)
+	. = ..()
+	if(isliving(AM))
+		var/mob/living/L = AM
+		playsound(loc, 'sound/effects/gib_step.ogg', HAS_TRAIT(L, TRAIT_LIGHT_STEP) ? 20 : 50, 1)
+	radiation_pulse(src, 500, 5) //MORE RADS
+
+/obj/effect/decal/nuclear_waste/attackby(obj/item/tool, mob/user)
+	if(tool.tool_behaviour == TOOL_SHOVEL)
+		radiation_pulse(src, 1000, 5) //MORE RADS
```

Comment:
This could easily be spammed without ever completing the do_after generating a stupid amount of radiation

Reply:
@ynot01 fixed